clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -446,7 +446,6 @@ public synchronized void close() {
     @SuppressWarnings("deprecation")
     @Override
     public synchronized void close(long timeout, TimeUnit unit) {
-        ensureNotClosed();
```

Comment:
This is a piggy-backed fix of MockConsumer: in the actual KafkaConsumer we do allow the `close()` call to be triggered multiple times, so there's no point checking this.